.github/workflows/tests.yml

```diff
@@ -1,43 +1,49 @@
-name: tests
+name: Tests
 
 on:
-  push: null
-  pull_request: null
+  pull_request:
+  push:
+    branches: [master]
 
 jobs:
-  tests:
-    name: tests
-    runs-on: "ubuntu-latest"
+  run:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        python-version: ["3.6", "3.7", "3.8", "3.9", "3.10"]
+        os: [windows-latest, ubuntu-latest, macos-latest]
+      fail-fast: false
+
     steps:
       - name: cancel previous runs
-        uses: styfle/cancel-workflow-action@0.6.0
+        uses: styfle/cancel-workflow-action@0.10.0
         with:
           access_token: ${{ github.token }}
 
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
 
-      - uses: conda-incubator/setup-miniconda@v2
+      - name: Setup Micromamba
+        uses: mamba-org/provision-with-micromamba@main
         with:
```

Comment:
I was thinking about the caching options but I guess that wouldn't help since no conda env yml file

Reply:
Probably. I only use cache for big envs, this one is not that heavy but we can try that in another PR I already added too much in this one.